### PR TITLE
ci(mk): update golangci-lint to v1.49.0

### PR DESCRIPTION
**Pull request by michaelbeaumont**
Well apparently this version blows up on memory too so we can just run it twice with different linters.

`contextcheck` works again and better than ever apparently so a lot of this PR is about fixing these errors.

Can anyone figure out https://github.com/kumahq/kuma/pull/4907/commits/538d3727c030a34fa888cb1873531e88f062a620?

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](/CONTRIBUTING.md#submitting-a-patch)?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: mk/check.mk [hunk 1]
```diff
@@ -24,7 +24,14 @@ shellcheck:
...
 golangci-lint: ## Dev: Runs golangci-lint linter
-	$(GOLANGCI_LINT_DIR)/golangci-lint run --timeout=10m -v
+	$(GOLANGCI_LINT_DIR)/golangci-lint run \
+		--disable-all \
+		--enable bodyclose,contextcheck,errcheck,gci,gocritic,gofmt,gomodguard,govet,importas,ineffassign,misspell,typecheck,unconvert,unparam,whitespace \
+		--timeout=10m -v
+	$(GOLANGCI_LINT_DIR)/golangci-lint run \
+		--disable-all \
+		--enable gosimple,staticcheck,unused \
+		--timeout=10m -v
```
Comment (bartsmykla) [inline mk/check.mk:34]: Just a question out of curiosity - why we are calling it twice instead of once? Is it because of timeout?

**Review by bartsmykla — APPROVED** [review]
lgtm

File: mk/check.mk (hunk 1 above)
Comment (michaelbeaumont) [reply mk/check.mk:34]: It's that golangci-lint runs OOM in ciecleci since `1.47.1`. I'll add a comment here as well

Comment (bartsmykla) [reply mk/check.mk:34]: got it

**Pull request merged**